mrp_production_replace_component/wizard/wiz_replace_component.py

```diff
@@ -0,0 +1,109 @@
+# © 2019 Akretion David BEAL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api, fields, _
+from odoo.exceptions import UserError
+
+
+class ReplaceComponent(models.TransientModel):
+    _name = 'replace.component.transient'
+
+    production_id = fields.Many2one(
+        comodel_name='mrp.production', required=True,
+        default=lambda s: s.env.context.get('active_id'))
+    remove_product_ids = fields.Many2many(
+        comodel_name='product.product', string='Products')
+    remove_product_domain_ids = fields.Many2many(
+        comodel_name='product.product', compute='_compute_remove_prd_domain',
+        help="Only required to store domain of available removable products.")
+    product_id = fields.Many2one(
+        comodel_name='product.product', string='Product',
+        domain=[('bom_line_ids', '!=', False)])
+    product_uom_qty = fields.Float(string='Quantity', default=1)
+    product_uom_id = fields.Many2one(
+        comodel_name='uom.uom', string='Unit', readonly=True)
+
+    @api.depends('production_id.move_raw_ids')
+    def _compute_remove_prd_domain(self):
+        prd_ids = [x.product_id.id for x in self.production_id.move_raw_ids]
+        for rec in self:
+            rec.remove_product_domain_ids = prd_ids
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.product_uom_id = self.product_id.uom_id or False
+
+    def apply_replacement(self):
+        self.ensure_one()
+        messages = []
+        if self.remove_product_ids:
+            remove_mess = self._remove_raw_materials()
+            if remove_mess:
+                messages.append(_("Component %s removed" % remove_mess))
+        if self.product_id:
+            self._append_raw_materials()
+            messages.append(_("Component '%s' added" % self.product_id.name))
+        self.production_id._adjust_procure_method()
+        self.production_id.move_raw_ids._action_confirm()
```

Comment:
Here, I think you should do self.production_id.move_raw_ids.filtered(lambda m: m.state == 'draft')
Because one part of the raw moves are already confirmed, you just want to confirm the one you just have created

Reply:
done